### ISSUE TITLE
[bug] `RainbowKitSiweNextAuthProvider` not working with NextAuth.js v5

### ISSUE THREAD

**Issue by edxeth** [rainbow-me/rainbowkit#1682]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

1.3.2

### wagmi Version

1.4.12

### Current Behavior

I'm thrown a runtime error when I try to use `RainbowKitSiweNextAuthProvider` with NextAuth.js v5:

![image](https://github.com/rainbow-me/rainbowkit/assets/105494645/7b0c2a4f-298d-4db2-85ea-c3526415b2b6)

The issue has appeared upon migration to v5. (it worked on v4)
Unwrapping `RainbowKitProvider` from `RainbowKitSiweNextAuthProvider` resolves the issue even though I'm not able to leverage Sign-in with Ethereum (siwe) then.

**@components/RainbowKit.tsx**:

![image](https://github.com/rainbow-me/rainbowkit/assets/105494645/09bf93fa-3e03-49e1-8eef-bd75a477fdbb)

**@/app/layout.tsx**:

![image](https://github.com/rainbow-me/rainbowkit/assets/105494645/35b74e8b-255a-49a4-b72e-0e4d298518fa)

My dependencies:

- @rainbow-me/rainbowkit: 1.3.2
- @rainbow-me/rainbowkit-siwe-next-auth: 0.3.4
- next-auth: 5.0.0-beta.4
- siwe: 2.1.4
- viem: 1.21.3
- wagmi: 1.4.12

**Comment by c0rdeiro** [comment]
I think it is not compatible with next-auth v5 yet
https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit-siwe-next-auth%400.3.3

**Comment by DanielSinclair** [comment]
@edxeth Currently the peer version is not supported and you should receive a package manager warning. Feel free to ping here when next-auth v5 is out of beta, and we can work on adding support!

**Comment by KosmosKey** [comment]
@edxeth The wagmi v2 and viem v2 support will be released very soon, but we do have a beta version that you can try out. Feel free to look at the migration guide here: https://beta.rainbowkit.com/guides/rainbowkit-wagmi-v2

Let me know if you find any issues 👍

**Comment by KeeganEad** [comment]
@KosmosKey with that will the `rainbowkit-siwe-next-auth` dependency be upgraded to `rainbowkit@2.x` ?

**Comment by KosmosKey** [comment]
> @KosmosKey with that will the `rainbowkit-siwe-next-auth` dependency be upgraded to `rainbowkit@2.x` ?

Yes that's correct. RainbowKit will be upgraded to `2.x`. Just make sure to update @rainbow-me/rainbowkit-siwe-next-auth to the latest version `0.4.x` once it's out.

**Comment by KosmosKey** [comment]
@edxeth Are you still facing this issue ? If yes do you have a small code example i can look at ?

**Comment by edxeth** [comment]
@KosmosKey is Rainbowkit v2 now supporting Next-Auth v5 (still in beta)?

**Comment by KosmosKey** [comment]
> @KosmosKey is Rainbowkit v2 now supporting Next-Auth v5 (still in beta)?

We don't support beta yet no. Are you able to use the latest non beta version for now and when the v5 launches we'll make sure to update it with RainbowKit.

**Comment by DanielSinclair** [comment]
Auth.js seems to be approaching a production release of v5 soon, so we will begin working on this. We are also evaluating other improvements to the Authentication/SIWE flow. Do you use SIWE alongside other auth methods in your next-auth configuration, or would you prefer if most of the boilerplate implementation is handled within RainbowKit's adapter package? I think we have the opportunity to make this much simpler for devs! Feedback or ideas are appreciated.

**Comment by c0rdeiro** [comment]
@DanielSinclair Although next auth does require a significant amount of boilerplate I think the priority should move away from siwe lib since it requires ethers still
I see you're working here #1997 on moving to the new viem siwe which already makes me very happy. Keep up the great work

**Comment by EF09FE** [comment]
any updates?

**Comment by c0rdeiro** [comment]
@EF09FE check the version [2.2.0](https://github.com/rainbow-me/rainbowkit/releases/tag/%40rainbow-me%2Frainbowkit%402.2.0) release notes

**Comment by Azleal** [comment]
Is there any demo using nextjs app router with rainbowkit and nextAuth-v5?  I've checked examples from https://github.com/rainbow-me/rainbowkit/tree/main/examples/ . none of these meet all the requirements above. thanks
